integration-libs/opf/base/components/opf-cta/opf-cta-scripts/opf-cta-scripts.service.ts

```diff
@@ -0,0 +1,263 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable, inject } from '@angular/core';
+import { CmsService, Product, isNotNullable } from '@spartacus/core';
+import { Order, OrderFacade, OrderHistoryFacade } from '@spartacus/order/root';
+import { Observable, from, of, throwError } from 'rxjs';
+import {
+  concatMap,
+  filter,
+  finalize,
+  map,
+  reduce,
+  switchMap,
+  take,
+} from 'rxjs/operators';
+
+import { OrderEntry } from '@spartacus/cart/base/root';
+import {
+  CmsPageLocation,
+  CtaScriptsLocation,
+  CtaScriptsRequest,
+  CtaScriptsResponse,
+  OpfDynamicScript,
+  OpfPaymentFacade,
+  OpfResourceLoaderService,
+} from '@spartacus/opf/base/root';
+import { CurrentProductService } from '@spartacus/storefront';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class OpfCtaScriptsService {
+  protected opfPaymentFacade = inject(OpfPaymentFacade);
+  protected orderDetailsService = inject(OrderFacade);
+  protected orderHistoryService = inject(OrderHistoryFacade);
+  protected opfResourceLoaderService = inject(OpfResourceLoaderService);
+  protected cmsService = inject(CmsService);
+  protected currentProductService = inject(CurrentProductService);
```

Comment:
[MAJOR]

IMHO the `OpfCtaScriptsService` has too much responsibilies. To become more cohesive and robust, IMHO it should only act as an orchestrator and injected various Strategy classes. Just choose the most `Applicable` one and execute it. Thanks to such a design, the `OpfCtaScriptsService` would not need to directly depend all those various services and facades from various domains (order, product service, ...). 

I suggest creating separate new Strategy classes (let's name them `OpfCtaScriptsResolver`s). Each such class would be responsible for one `CmsPageLocation` and its `ScriptLocation`, as well as injecting and using specific services (e.g., `orderFacade` OR `currentProductService`  etc.) to get necessary data to run a script. Strategy classes should be multi-provided, and only one of them should be chosen in a "contest" (e.g., using the `Applicable` interface and `resolveApplicable()` utility) by the `OpfCtaScriptsService`. 

Please let me know if it makes sense. I'm happy to elaborate more, if needed.

Reply:
I can see now we've splitted out the page-specific implementations to 2 separate services - OpfCtaDynamicService and OpfCtaStaticService. Thanks to that, now the main OpfCtaScripts service bacame more focused on being just an orchestrator.

So I'm making this comment as OUTDATED